lib/simplecov/filter.rb

```diff
@@ -63,11 +65,23 @@ def matches?(source_file)
   end
 
   class ArrayFilter < SimpleCov::Filter
-    # Returns true if any of the file paths passed in the given array matches the string
-    # configured when initializing this Filter with StringFilter.new(['some/path', 'other/path'])
+    def initialize(filter_argument)
+      filter_argument.map! do |arg|
+        if arg.is_a?(SimpleCov::Filter)
+          arg
+        else
+          Filter.class_for_argument(arg).new(arg)
```

Comment:
Yah sounds good. I'll make a factory function that instantiates based the appropriate class type, or does nothing if it's given a filter.